static/app/components/events/interfaces/crashContent/exception/useSourceMapDebug.tsx

```diff
@@ -0,0 +1,193 @@
+
+import type {ExceptionValue, Frame, Organization, PlatformType} from 'sentry/types';
+import {defined} from 'sentry/utils';
+import {QueryKey, useQueries, useQuery, UseQueryOptions} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+
+interface BaseSourceMapDebugError {
+  message: string;
+  type: SourceMapProcessingIssueType;
+}
+
+interface UnknownErrorDebugError extends BaseSourceMapDebugError {
+  type: SourceMapProcessingIssueType.UNKNOWN_ERROR;
+}
+interface MissingReleaseDebugError extends BaseSourceMapDebugError {
+  type: SourceMapProcessingIssueType.MISSING_RELEASE;
+}
+interface MissingUserAgentDebugError extends BaseSourceMapDebugError {
+  data: {version: string};
+  type: SourceMapProcessingIssueType.MISSING_USER_AGENT;
+}
+interface MissingSourcemapsDebugError extends BaseSourceMapDebugError {
+  type: SourceMapProcessingIssueType.MISSING_SOURCEMAPS;
+}
+interface UrlNotValidDebugError extends BaseSourceMapDebugError {
+  data: {absValue: string};
+  type: SourceMapProcessingIssueType.URL_NOT_VALID;
+}
+interface PartialMatchDebugError extends BaseSourceMapDebugError {
+  data: {insertPath: string; matchedSourcemapPath: string};
+  type: SourceMapProcessingIssueType.PARTIAL_MATCH;
+}
+
+export type SourceMapDebugError =
+  | UnknownErrorDebugError
+  | MissingReleaseDebugError
+  | MissingUserAgentDebugError
+  | MissingSourcemapsDebugError
+  | UrlNotValidDebugError
+  | PartialMatchDebugError;
+
+export interface SourceMapDebugResponse {
+  errors: SourceMapDebugError[];
+}
+
+export enum SourceMapProcessingIssueType {
+  UNKNOWN_ERROR = 'unknown_error',
+  MISSING_RELEASE = 'no_release_on_event',
+  MISSING_USER_AGENT = 'no_user_agent_on_release',
+  MISSING_SOURCEMAPS = 'no_sourcemaps_on_release',
+  URL_NOT_VALID = 'url_not_valid',
+  PARTIAL_MATCH = 'partial_match',
+}
+
+const sourceMapDebugQuery = ({
+  orgSlug,
+  projectSlug,
+  eventId,
+  frameIdx,
+  exceptionIdx,
+}: UseSourceMapDebugProps): QueryKey => [
+  `/projects/${orgSlug}/${projectSlug}/events/${eventId}/source-map-debug/`,
+  {
+    query: {
+      frame_idx: `${frameIdx}`,
+      exception_idx: `${exceptionIdx}`,
+    },
+  },
+];
+
+interface UseSourceMapDebugProps {
+  eventId: string;
+  exceptionIdx: number;
+  frameIdx: number;
+  orgSlug: string;
+  projectSlug: string;
+}
+
+export type StacktraceFilenameQuery = {filename: string; query: UseSourceMapDebugProps};
+
+export function useSourceMapDebug(
+  props?: UseSourceMapDebugProps,
+  options: Partial<UseQueryOptions<SourceMapDebugResponse>> = {}
+) {
+  return useQuery<SourceMapDebugResponse>(props ? sourceMapDebugQuery(props) : [''], {
+    staleTime: Infinity,
+    retry: false,
+    refetchOnWindowFocus: false,
+    notifyOnChangeProps: ['data'],
+    ...options,
+    enabled: !!options.enabled && defined(props),
+  });
+}
+
+export function useSourceMapDebugQueries(props: UseSourceMapDebugProps[]) {
+  const api = useApi({persistInFlight: true});
```

Comment:
mind explanation why we are setting this? the explanation in the `useApi` hook is not clear to me

Reply:
i think we have to persist them with useQuery since it doesn't know why it was cancelled https://github.com/getsentry/sentry/pull/42123